modules/core/middleware/middleware.go

```diff
@@ -0,0 +1,128 @@
+package middleware
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/v3/modules/core/keeper"
+)
+
+var _ tx.Handler = ibcTxHandler{}
+
+type ibcTxHandler struct {
+	k    *keeper.Keeper
```

Comment:
Why is this a pointer?

Reply:
I have no preference, but typically keepers are not pointers. I haven't checked into why it needs to be a pointer in app.go

Reply:
Hi Colin,

I forget the reason precisely, but I do know that I broke osmosis quite badly by not making it a pointer in app.go in the short-lived osmosis v5 release.